### ISSUE TITLE
CPP/ConstMe Whisper models folder

### ISSUE THREAD

**Issue by vivadavid** [SubtitleEdit/subtitleedit#6807]
Hi,

As you can download the Whisper models directly from Huggin Face, would it be possible to make Subtitle Edit read these models from an external folder? This way, users would be able to use the same models with different frontends, avoiding having to download the same files over and over again.

Thanks for considering the request.

**Comment by Clarencezzh** [comment]
I have a similar problem that I want to put the whisper and vosk, etc. into the app folder, but not C:\Users\users\AppData\Roaming\Subtitle Edit. Or the C drive may become too bloated.

**Comment by uckthis** [comment]
I was thinking it too. There are a lot of fine-tuned models available in targeted languages on Hugging Face, and if we could just download the BIN file and place it in the Models folder, that would help a lot. Currently, it only allows the default models.

**Comment by niksedk** [comment]
> I was thinking it too. There are a lot of fine-tuned models available in targeted languages on Hugging Face, and if we could just download the BIN file and place it in the Models folder, that would help a lot. Currently, it only allows the default models.

The `.bin` files are in the SE Data folder, in `Whisper\Models`

**Comment by uckthis** [comment]
Yes, I understand that. I meant if SE could read the BIN files other than the default models. I downloaded a Whisper model BIN file today from Hugging Face, placed it in the Models folder, but it didn't show up in Whisper UI. Or, do we need some extra files for that BIN to work?
 https://huggingface.co/Shubham09/whisper-small-French/tree/main
Can we not use the BIN file from this? 
 I'm not computer savvy, so I may be way off the mark here.

**Comment by niksedk** [comment]
If you can make it work with the command line, I can take a closer look at allowing custom bin models...

Did not work for me:

````

Message: Calling whisper (CPP) with : C:\Tools\SE\Whisper\main.exe --language fr --model "C:\Tools\SE\Whisper\Models\pytorch_model.bin" --output-srt --print-progress  "C:\Users\nikse\AppData\Local\Temp\d799e82d-8711-4ed4-9e7a-71bd51c0d19f.wav"

error: failed to initialize whisper context

whisper_init: failed to load model

whisper_model_load: invalid model data (bad magic)

````

**Comment by darnn** [comment]
If I recall correctly, you need to download the entire HF model (not just the bin file), and convert it using this: https://github.com/ggerganov/whisper.cpp/tree/master/models#fine-tuned-models

I did it once months ago and it did work (with whisper.cpp itself, at least, I think Const-Me didn't exist yet then) but it took a lot of effort to figure out Python's cryptic error messages (ffmpeg.exe wasn't in my path, apparently). But I'm sure it would go easier for someone who knows what they're doing.

**Comment by uckthis** [comment]
I did not use this locally. I used Hugging Face https://huggingface.co/Shubham09/whisper-small-French to run it and transcribe a sample file.

**Comment by Purfview** [comment]
And similar issue: https://github.com/SubtitleEdit/subtitleedit/issues/6698

**Comment by niksedk** [comment]
> I have a similar problem that I want to put the whisper and vosk, etc. into the app folder, but not C:\Users\users\AppData\Roaming\Subtitle Edit. Or the C drive may become too bloated.

You can always use the portable version, and unpack to something like D:\Tools\SE

**Comment by timminator** [comment]
> I have a similar problem that I want to put the whisper and vosk, etc. into the app folder, but not C:\Users\users\AppData\Roaming\Subtitle Edit. Or the C drive may become too bloated.

I came across this problem aswell today and found another solution. I moved the models folder to my desired location and created a symbolic link. Worked perfectly fine on my end and i could free up my C drive.
I did this for the Purfview-Faster-Whisper models stored in "C:\Users\YourUserName\AppData\Roaming\Subtitle Edit\Whisper\Purfview-Whisper-Faster".

Just

1. Move your models folder to your desired location
2. Create a symbolic link by opening a cmd window and running the following command:
    `mklink /J "C:\Users\YourUserName\AppData\Roaming\Subtitle Edit\Whisper\Purfview-Whisper-Faster\_models" "D:\Users\YourDesiredLocation\_models"'
    
I hope it helps someone else aswell!
